### ISSUE TITLE
Testing the F1 indicator

### ISSUE THREAD

**Issue by xupinggl** [HighwayWu/FOCAL#9]
First of all, thank you very much for sharing the code. I tested the results with the pre-trained weights, and the other metrics are almost the same, but the results of the F1 metrics are not quite the same as the method of calculating them used by the IF-OSN paper, I'd like to ask about the difference between the two methods of calculating the F1 metrics? Thank you very much.
![1](https://github.com/HighwayWu/FOCAL/assets/102807353/106510c6-36c0-47fe-b4d0-3019b5d418e1)
![2](https://github.com/HighwayWu/FOCAL/assets/102807353/7ae5069e-5391-41ed-a280-d5420afb3597)

**Comment by xupinggl** [comment]
![image](https://github.com/HighwayWu/FOCAL/assets/102807353/ab52de8e-3b79-423f-962f-0c897d8c9fb8)

**Comment by HighwayWu** [comment]
Thanks for your question. Actually, the way we calculated the F1 metric in FOCAL and IF-OSN is different. In IF-OSN, we used an existing metric from [Tianchi Competition](https://github.com/HighwayWu/Tianchi-FFT2). However, in FOCAL, to be more consistent with the evaluation methods of algorithms such as TruFor, we adopted the macro-based F1 calculation method. For more detailed instructions, please refer to the "average" parameter of the [f1_score](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.f1_score.html) in sklearn library.

**Comment by xupinggl** [comment]
Thank you for your answer, I got it.
